Casks/thunder.rb

```diff
@@ -10,9 +10,12 @@
   homepage "https://www.xunlei.com/"
 
   livecheck do
-    url :homepage
-    strategy :page_match
-    regex(/thunder_(\d+(?:\.\d+)*)\.dmg/i)
+    url "https://static-xl9-ssl.xunlei.com/json/mac_preferences.json"
+    strategy :page_match do |page|
+      version = page[/appcast_(\d+(?:\.\d+)*)\.xml/i, 1]
+      version_page = Homebrew::Livecheck::Strategy.page_content("https://down.sandai.net/mac/thunder/appcast_#{version}.xml")
+      version_page[:content].scan(/thunder[_-](\d+(?:\.\d+)*)\.dmg/i).flatten
+    end
```

Comment:
As mentioned in previous comment, let's try using simpler livecheck for now. If it breaks, we can add extra logic to deal with it:
```suggestion
    regex(/appcast[._-]v?(\d+(?:\.\d+)+)\.xml/i)
```

Also some regex-style changes to align with Formula-side (e.g. https://docs.brew.sh/Formula-Cookbook#livecheck-blocks) like using more lenient `[._-]` for in-between characters, etc.